packages/atomic/src/components/result-template-components/atomic-size-condition/atomic-size-condition.tsx

```diff
@@ -0,0 +1,150 @@
+import {Element, Component, Prop, State} from '@stencil/core';
+
+type Criterion = (width: number, height: number) => boolean;
+
+/**
+ * Displays children only if their parent meets the defined size constraints.
+ */
+@Component({
+  tag: 'atomic-size-condition',
+  styleUrl: 'atomic-size-condition.pcss',
+  shadow: false,
+})
+export class AtomicSizeCondition {
+  private resizeObserver!: ResizeObserver;
+
+  @Element() private host!: HTMLElement;
+
+  @State() private shouldRender!: boolean;
+
+  /**
+   * The minimum width (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumWidthExclusive?: string;
+  /**
+   * The minimum width (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumWidthInclusive?: string;
+  /**
+   * The maximum width (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumWidthExclusive?: string;
+  /**
+   * The maximum width (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumWidthInclusive?: string;
+  /**
+   * The minimum height (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumHeightExclusive?: string;
+  /**
+   * The minimum height (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumHeightInclusive?: string;
+  /**
+   * The maximum height (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumHeightExclusive?: string;
+  /**
+   * The maximum height (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumHeightInclusive?: string;
+
+  constructor() {
+    this.shouldRender = this.elementMeetsCriteria;
+  }
+
+  public componentDidLoad() {
+    this.resizeObserver = new ResizeObserver(
+      () => (this.shouldRender = this.elementMeetsCriteria)
+    );
+    this.resizeObserver.observe(this.parent);
+  }
+
+  public componentDidUnLoad() {
+    this.resizeObserver.disconnect();
+  }
+
+  public componentDidRender() {
+    this.host.style.display = this.shouldRender ? '' : 'none';
+  }
+
+  private convertToPixels(size: string) {
+    const temporaryElement = document.createElement('div');
+    temporaryElement.style.position = 'absolute';
+    temporaryElement.style.width = size;
+    this.parent.appendChild(temporaryElement);
+    const {offsetWidth} = temporaryElement;
+    temporaryElement.remove();
+    return offsetWidth;
+  }
+
+  private get parent() {
+    return this.host.parentElement!;
+  }
```

Comment:
If you prefer not to use the parent element, the alternative would be to use the host element and render an additional child inside this component which would be hidden/shown instead of the host.